src/utils/proxy.js

```diff
@@ -247,7 +242,7 @@ const serveRedirect = async function ({ req, res, proxy, match, options }) {
     }
 
     if (isFunction(options.functionsPort, req.url)) {
-      req.headers['x-netlify-original-pathname'] = reqUrl.pathname
+      req.headers['x-netlify-original-pathname'] = reqToURL(req, originalURL).pathname
```

Comment:
We need the path, but without the query parameters and hash.
One way to do this using `req.url` (which includes all of those but not the origin) is to parse it with `new URL()`. This re-uses the `new URL()` logic from above to do this.